kubernets/kubectl/cluster.go

```diff
@@ -65,34 +67,54 @@ func (c *Cluster) DeleteConfig() error {
 //nolint:forcetypeassert
 func (c *Cluster) Namespaces(ctx context.Context) []string {
 	return c.kubectl.cache.Get(c.name+"-namespaces", func() any {
-		if out, err := shell.New(ctx, c.l, "kubectl",
+		sh := shell.New(ctx, c.l, "kubectl",
 			"get", "namespaces",
-			"-o", "jsonpath='{.items[*].metadata.name}'",
-		).
-			Env(c.Env()).
-			Output(); err != nil {
+			"-o", "jsonpath='{.items[*].metadata.name}'").
+			Env(c.Env())
+
+		if c.kubectl.authTokenProvider != nil {
+			token, err := c.kubectl.authTokenProvider(ctx, c.Name())
+			if err != nil {
+				c.l.Warn(err.Error())
+				return []string{}
+			}
+			sh.Args("--token", token)
+		}
+
+		out, err := sh.Output()
+		if err != nil {
 			c.l.Warn(err.Error())
 			return []string{}
```

Comment:
any reason not to use nil here?

Reply:
yes, this is within the cache method and it would fail due to type casting at the end

Reply:
I guess nil is not cache-able